VCC2018 recipe

unilight opened the pull request:
Recipe for training VQVAE on VCC2018.

1. I found that the config files for VCC2020 are gone (due to rebase?) so I added them.

2. Also, I found that `*.yml` files are ignored.
https://github.com/k2kobayashi/crank/blob/master/egs/vaevc/.gitignore#L5
Is there a specific reason?

3. About your comment on the recipe design, 
> We supposed to share run.sh over all recipes (i.e., use symbolic link) because of reducing maintenance const for further recipes. Are there any difficulty to do that? If so, it will require to modify template/run.sh.

Currently, I am okay with sharing recipes. Creating new `run.sh` files is just a habit from ESPnet. On the other hand, creating a new `run.sh` for every dataset can make the design more flexible. Flexibility might become an issue as we develop more models and train on more datasets. But anyway this is all about personal choice.

k2kobayashi committed:
rebase commit for release

k2kobayashi committed:
fix scp generation

k2kobayashi committed:
Merge pull request #38 from k2kobayashi/scp_bugfix

fix scp generation

unilight committed:
vcc2018; gen-scp

unilight committed:
gen-scp

unilight committed:
merge

unilight committed:
configs

unilight committed:
Merge branch 'master' of https://github.com/k2kobayashi/crank

unilight committed:
Merge branch 'master' into vcc2018

unilight committed:
minor fix

unilight committed:
remove unnecessary files

unilight committed:
minor fix

k2kobayashi commented:
For the repository policy, It should keep our templates to share all recipe.
By retaining such structure, when we add new features such as neural vocoder, it is straightforward to implement it because we just modify and test just once.
So, when creating a new recipe, we just modify ```egs/vaevc/<recipe>/{conf,local,data}``` .

Of course, this constraint makes flexibility degradation. But, when it happens, we can create new method (i.e., move on egs/<new_feature> directory). 
In terms of contribution for recipe creation, it'll describe their usernames on REAMD.md.

unilight committed:
Merge branch 'master' of https://github.com/k2kobayashi/crank into vcc2018

unilight committed:
change to use run.sh in template

unilight commented:
I changed to use `template/run.sh`. Also, I changed so that `download.sh` takes `downloaddir` as input instead of `wavdir` so we can save other files, ex. transcriptions. This is for, ex. ASR evaluation.
It should be okay to merge now.

k2kobayashi commented on egs/vaevc/template/run.sh:
```diff
@@ -24,7 +24,7 @@ n_jobs=10 # number of parallel jobs
 
 # directory settings
-wavdir=downloads/wav # directory to save downloaded wav files
+downloaddir=        # directory to save downloaded wav files
```
When $downloaddir is not defined, $wavdir is placed in ```/wav``` not ```./wav``` isn't it?

unilight replied:
I see. I will specify a default value for `downloaddir`.

k2kobayashi commented on egs/vaevc/vcc2018v1/conf/default.yml:
```diff
@@ -0,0 +1,130 @@
+# feature setting
```
Is it ok to make default.yml as symbolic link from ```template/conf```? Maybe you need to overwrite it by ```vcc2020v1/conf/default.yml```.

unilight replied:
Sorry, I don't really understand... the current design is to put separate configs for each dataset, right?

k2kobayashi replied:
Sorry, it requires to change feature extraction settings in each recipe.
I thought the word of default should be default for vaevc not each recipe, but it's my misunderstandings.

unilight replied:
I see. Or shall we use more meaningful names? For example, if we want to compare different network types or features, we can name it to something like `mlfb.yml`, `dilatedcnn.yml`?

k2kobayashi replied:
That's good idea. ```${feat_type}_${trainer_type}.yml``` like ```{mlfb}_{vqvae}.yml``` is fine.

k2kobayashi commented on egs/vaevc/vcc2018v1/conf/slurm.conf:
```diff
@@ -0,0 +1,9 @@
+command sbatch --export=PATH  --ntasks-per-node=1
```
Is it ok to as symbolic link? I did not try.

unilight replied:
Same as above.

k2kobayashi reviewed (commented):
Thanks! I have a few comments related to create a new recipe. Could you include such as modification to this PR?

unilight committed:
default downloaddir name; rename conf file

k2kobayashi commented on egs/vaevc/template/run.sh:
```diff
@@ -45,6 +45,7 @@ eval_speakers=""  # evaluation speaker
 set -eu # stop when error occured and undefined vars are used
 
 mkdir -p "${expdir}"
+wavdir=${downloadir}/wav
```
downloadir
->
downloaddir

k2kobayashi commented on egs/vaevc/template/run.sh:
```diff
@@ -53,7 +54,7 @@ logdir=${datadir}/log; mkdir -p ${logdir}
     echo "stage 0: download dataset and generate scp"
     ${train_cmd} "${logdir}/download.log" \
-        local/download.sh --wavdir "${wavdir}"
+        local/download.sh "${downloaddir}"
```
```
local/download.sh "${downloaddir}"
->
local/download.sh \
  --downloaddir "${downloaddir}"
```
For kaldi-style arguments of shell scripts, it uses parse_options.sh which enables to automatically deploy arguments to named variable. You also need to modify ```local/download.sh```

k2kobayashi commented on egs/vaevc/vcc2018v1/local/download.sh:
```diff
@@ -0,0 +1,68 @@
+# Distributed under terms of the MIT license.
+#
+
+db_root=$1
```
```
db_root -> downloaddir
```
need to modify all $db_root variables.

k2kobayashi reviewed (commented):
I ran and checked download scripts. followings are some comments.

k2kobayashi committed:
add contribution to README.md

k2kobayashi commented on egs/vaevc/vcc2018v1/local/download.sh:
```diff
@@ -0,0 +1,68 @@
+    
+    # rename the folders
+    cd ${wavdir}
+    rename "s/VCC2//g" *
```
I do not have ```rename``` command. Is it possible to change it by substitute?

unilight replied:
Can I know what platform you are using? I didn't know that `rename` is platform-specific...

k2kobayashi replied:
Ubuntu 18.04.4 LTS as well as Mac OSX.

unilight replied:
Okay. I changed to use `mv`.

unilight committed:
argument change

unilight committed:
Merge branch 'vcc2018' of https://github.com/unilight/crank into vcc2018

unilight committed:
change rename to mv

k2kobayashi commented:
Thanks and good job @unilight. I merged.
